wavelength-plugins/friends.js

```diff
@@ -0,0 +1,117 @@
+/*
+ * FC Friends
+ * By WGC
+ */
+'use strict';
+
+function genFC() {
+	const randomCode = () => Math.floor(Math.random() * 89999999) + 10000000;
+	let currentCodes = Db.fc.keys().map(curKey => Db.fc.get(curKey));
+	let uniqueCode = `WL${randomCode()}`;
+	while (currentCodes.some(curCode => uniqueCode === curCode)) {
+		uniqueCode = `WL${randomCode()}`;
+	}
+	return uniqueCode;
+}
+const FC = {
+	getFC: function (user) {
+		user = toId(user);
+		let fcget = Db.fc.get(user);
+		if (!fcget || fcget === null) return user.reply('You don\'t have a friendcode. Use /fc make to make one.');
+		return fcget;
+	},
+	addFC: function (user, fc) {
+		user = toId(user);
+		Db.fc.set(user, fc);
+		return user.reply('Your friend code has been added to the system. Give to others to add you as a friend!');
+	},
+};
+const Friend = {
+	addFriend: function (target, fc, user) {
+		target = toId(target);
+		fc = fc.toLowerCase();
+		let thisFC = FC.getFC(target).toLowerCase();
+		if (fc === thisFC) {
+			let curFriends = Db.friend.get(user, []);
+			curFriends.push(target);
+			Db.friend.set(user, curFriends);
+			user.popup(`You have added ${target} as a friend.`);
+			return;
+		} else {
+			user.popup(`${fc} is not the right friendcode for ${target}.`);
+			return;
+		}
+	},
+	removeFriend: function (target, user) {
+		target = toId(target);
+		if (!Db.friend.get(user).includes(target)) return this.errorReply('You are not friends with this user.');
```

Comment:
This line gets handled in lines 49-52 because we need to first ensure that `user` is the userid, then that `Db.friend.get(user)` returns a non-null value, etc.; this line can be deleted.

Reply:
(Also, `Friend` still doesn't have the same `this` value as a command, because it's `this` value is the `Friend` object itself, not an instance of `Chat.CommandContext`.)